### PR TITLE
 Add Jenkins cert-based auth in prow 

### PR DESCRIPTION
**Pull request by kargakis**
Adds certificate based authentication and CSRF protection in the Jenkins client
which effectively enables the operator to work with locked down Jenkins masters.

### CODE REVIEW

File: prow/cmd/jenkins-operator/main.go [hunk 1]
```diff
@@ -91,8 +96,16 @@ func main() {
 		ac.BearerToken = &jenkins.BearerTokenAuthConfig{
 			Token: token,
 		}
+	} else if *certFile != "" && *keyFile != "" && *caCertFile != "" {
```
Comment (kargakis) [inline prow/cmd/jenkins-operator/main.go]: @dobbymoodge I have a feeling that the certs are complementary to doing basic auth. Is that correct?

**Comment by dobbymoodge** [comment]
not certain the CACert is *necessary* on the client side, but it shouldn't hurt. I need to take a closer look at what the code is doing with it.

**Comment by kargakis** [comment]
I can make it optional if it's not necessary.

**Comment by kargakis** [comment]
I made the CA cert optional.

**Comment by kargakis** [comment]
/hold cancel

@stevekuznetsov @dobbymoodge ptal, I have also added csrf protection in this PR

File: prow/cmd/jenkins-operator/main.go [hunk 2]
```diff
@@ -94,8 +102,19 @@ func main() {
 	} else {
 		logger.Fatal("An auth token for basic or bearer token auth must be supplied.")
 	}
+	var tlsConfig *tls.Config
+	if *certFile != "" && *keyFile != "" {
```
Comment (stevekuznetsov) [inline prow/cmd/jenkins-operator/main.go]: seems like we should just validate that if any of the three (cert, key, cacert) are present that all three are?

**Comment by dobbymoodge** [comment]
@stevekuznetsov LGTM, as far as I can understand the code. If it works the way we expect, then I /lgtm

**Pull request merged**